torch/csrc/jit/passes/erase_detach.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "torch/csrc/jit/ir.h"
+
+namespace torch { namespace jit {
+
+// Erase detach. This is necessary for export to ONNX, since ONNX
+// does not support training. This is safe because, again,
+// ONNX does not suppor ttracing.
+TORCH_API void EraseDetach(const std::shared_ptr<Graph>& graph);
```

Comment:
You don't need an entire pass to do this. Here is how it is done for a similar op, clone in `symbolics.py`:

```
# ignore clone operators that are inserted by PyTorch autograd
def clone(g, input):
    return input
```